linera-execution/src/execution.rs

```diff
@@ -225,11 +221,11 @@ where
             .with_state_and_grant(&mut self.system, cloned_grant.as_mut())
             .await?
             .balance()?;
-        let controller = ResourceController {
-            policy: resource_controller.policy.clone(),
-            tracker: resource_controller.tracker,
-            account: initial_balance,
-        };
+        let controller = ResourceController::new(
```

Comment:
We could omit the `account` parameter from `new`, and instead have a `with_account` function.

Reply:
https://github.com/linera-io/linera-protocol/issues/3993